lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -274,14 +274,42 @@ function _campaign_resource_signup($nid, $values) {
   if (!isset($values['source'])) {
     $values['source'] = NULL;
   }
+  // @TODO: handle passing this to Rogue so it is not lost (and add to Phoenix documentation which is missing this)
   if (!isset($values['transactionals'])) {
     $values['transactionals'] = TRUE;
   }
   if (DOSOMETHING_SIGNUP_LOG_SIGNUPS) {
     watchdog('dosomething_api', '_campaign_resource_signup values:' . json_encode($values));
   }
-  // transactionals = FALSE will result in neither email or SMS transactional messages being sent.
-  return dosomething_signup_create($nid, $values['uid'], $values['source'], NULL, $values['transactionals']);
+
+  $headers = getallheaders();
+
+  // Check to see if the request came from Rogue and if we should be sending it there
+  // $headers['X-Request-Id'] will only be set if the request is coming from Rogue
+  if ( !isset($headers['X-Request-Id']) && variable_get('rogue_signup_collection', FALSE)) {
+    // Pass the campaign_id
+    $values['campaign_id'] = $nid;
+
+    // Get the run to pass
+    $user = user_load($values['uid']);
+    $run = dosomething_helpers_get_current_campaign_run_for_user($nid, $user);
+
+    // Post the signup to Rogue
+    $rogue_signup = dosomething_rogue_send_signup_to_rogue($values);
+
+    // Store the reference to the Rogue signup
+    return dosomething_rogue_check_sid_and_store_ref($rogue_signup);
+  } else {
+    // Increment transaction id and log that the request has been received with new transaction id.
+    $transaction_id_parts = explode('-', $headers['X-Request-Id']);
+    $transaction_id_base = $transaction_id_parts[0];
+    $incremented_step = $transaction_id_parts[1] + 1;
+    $incremented_transaction_id = $transaction_id_base . '-' . $incremented_step;
+
+    watchdog('PhoenixTransactionBridge', 'Request received. Transaction ID: !incremented_transaction_id', ['!incremented_transaction_id' => $incremented_transaction_id], WATCHDOG_INFO);
+    // transactionals = FALSE will result in neither email or SMS transactional messages being sent.
```

Comment:
Yeah it seems like that's one of the bigger things we need to tackle in order to turn off Phoenix. I guess Rogue will still have to handle ferrying this value around in the meantime though